src/libs/duplicate.c

```diff
@@ -0,0 +1,360 @@
+    gtk_box_pack_start(GTK_BOX(hb), tb, FALSE, FALSE, 5);
+    gtk_box_pack_start(GTK_BOX(hb), lb, FALSE, FALSE, 0);
+    gtk_box_pack_start(GTK_BOX(hb), bt, FALSE, FALSE, 0);
+
+    gtk_box_pack_start(GTK_BOX(d->duplicate_box), hb, FALSE, FALSE, 0);
+  }
+  sqlite3_finalize (stmt);
+  
+  gtk_widget_show_all(d->duplicate_box);
+}
+
+static void _lib_duplicate_mipmap_updated_callback(gpointer instance, dt_lib_module_t *self)
+{
+  dt_lib_duplicate_t *d = (dt_lib_duplicate_t *)self->data;
+  gtk_widget_queue_draw (d->duplicate_box);
+  dt_control_queue_redraw_center();
+}
+
+void gui_init(dt_lib_module_t *self)
+{
+  /* initialize ui widgets */
+  dt_lib_duplicate_t *d = (dt_lib_duplicate_t *)g_malloc0(sizeof(dt_lib_duplicate_t));
+  self->data = (void *)d;
+  
+  d->imgid = 0;
+  
+  self->widget = gtk_box_new(GTK_ORIENTATION_VERTICAL, DT_PIXEL_APPLY_DPI(5));
+  gtk_widget_set_name(self->widget, "duplicate-ui");
+  
+  GtkWidget *sw = gtk_scrolled_window_new(NULL, NULL);
+  gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(sw), GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
+  gtk_scrolled_window_set_min_content_height(GTK_SCROLLED_WINDOW(sw), DT_PIXEL_APPLY_DPI(300));
+  d->duplicate_box = gtk_box_new(GTK_ORIENTATION_VERTICAL, 0);
+
+  GtkWidget *hb = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
+  GtkWidget *bt = gtk_label_new(_("existing duplicates"));
+  gtk_box_pack_start(GTK_BOX(hb), bt, FALSE, FALSE, 0);
+  bt = dtgtk_button_new(dtgtk_cairo_paint_plusminus, CPF_ACTIVE | CPF_STYLE_FLAT | CPF_DO_NOT_USE_BORDER);
+  g_object_set(G_OBJECT(bt), "tooltip-text", _("create a 'virgin' duplicate of the image without any developpement"), (char *)NULL);
+  g_signal_connect(G_OBJECT(bt), "button-press-event", G_CALLBACK(_lib_duplicate_new_clicked_callback), self);
+  gtk_box_pack_end(GTK_BOX(hb), bt, FALSE, FALSE, 0);
+  bt = dtgtk_button_new(dtgtk_cairo_paint_multiinstance, CPF_STYLE_FLAT | CPF_DO_NOT_USE_BORDER);
+  g_object_set(G_OBJECT(bt), "tooltip-text", _("create a duplicate of the image with same history stack"), (char *)NULL);
+  g_signal_connect(G_OBJECT(bt), "button-press-event", G_CALLBACK(_lib_duplicate_duplicate_clicked_callback), self);
+  gtk_box_pack_end(GTK_BOX(hb), bt, FALSE, FALSE, 0);
+
+
+  /* add duplicate list and buttonbox to widget */
+  gtk_box_pack_start(GTK_BOX(self->widget), hb, FALSE, FALSE, 0);
+  gtk_container_add(GTK_CONTAINER(sw), d->duplicate_box);
+  gtk_box_pack_start(GTK_BOX(self->widget), sw, FALSE, FALSE, 0);  
+
+  gtk_widget_show_all(self->widget);
+  
+  dt_control_signal_connect(darktable.signals, DT_SIGNAL_DEVELOP_IMAGE_CHANGED, G_CALLBACK(_lib_duplicate_init_callback), self);
+  dt_control_signal_connect(darktable.signals, DT_SIGNAL_DEVELOP_INITIALIZE, G_CALLBACK(_lib_duplicate_init_callback), self);
+  dt_control_signal_connect(darktable.signals, DT_SIGNAL_DEVELOP_MIPMAP_UPDATED, G_CALLBACK(_lib_duplicate_mipmap_updated_callback), (gpointer)self);
+}
```

Comment:
You probably also want to listen to `DT_SIGNAL_COLLECTION_CHANGED` or whatever gets raised when a new image gets loaded to update your list.